src/lib/bitband.fth

```diff
@@ -0,0 +1,25 @@
+\ Bit-banding maps a complete word of memory onto a single bit in the bit-band
+\ region. For example, writing to one of the alias words sets or clears the
+\ corresponding bit in the bit-band region. This enables every individual bit in
+\ the bit-banding region to be directly accessible from a word-aligned address
+\ using a single LDR instruction. It also enables individual bits to be toggled
+\ without performing a read-modify-write sequence of instructions.
+
+1 #28 shift 1- constant BITBAND.BASEMASK_               \ covers the offset
+BITBAND.BASEMASK_ invert constant BITBAND.BASEMASK      \ covers only the page
+$2000000 constant BITBAND.OFFSET                        \ bit_word_offset is the position of the target bit in the bit-band memory region.
+
+: BITBAND ( addr bit -- aliasaddress ) 
+    $4 * \ offest caused by the bit
+    swap
+    dup                     \ we need to split the addr is two parts
+    BITBAND.BASEMASK and    \ get the bit_band_base - the starting address of the alias region. 
+    swap
+    BITBAND.BASEMASK_       \ get the byte_offset - the number of the byte in the bit-band region that contains the targeted bit.
+    and $20 *
+    + +
+    BITBAND.OFFSET +        \ add the bit_word_offset - the position of the target bit in the bit-band memory region.
+;
+
```

Comment:
```
: bitband  ( bit adr -- aliasadr )
   dup 5 lshift                      ( bit adr byte-offset )   \ Spread low part of address; high bits will be shifted out
   swap $f000.0000 and or  ( bit adr' )                     \ Keep high nibble of address
   $0200.0000 or                  ( bit adr' )                    \ Offset to bitband alias space
   swap la+                          ( aliasadr )                    \ Merge bit number as 32-bit word offset
;
```
The above is smaller, faster and more direct.  There is no value in declaring the masks as constants because they are fixed for this architecture.  The value of having symbolic names is nullified by the confusion caused by the similarity of the names that differ only in a trailing '_', which has no mnemonic value.

But, all that said, if one is to use the bitbanding feature, why not just define the bit object address within the bitband region directly, avoiding the address,bit# representation entirely?

```
$4222.0234 constant mybit
1 mybit l!
```

Reply:
Fixing the formatting:

```
: bitband  ( bit adr -- aliasadr )
   dup 5 lshift            ( bit adr byte-offset )  \ Spread low part of address; high bits will be shifted out
   swap $f000.0000 and or  ( bit adr' )             \ Keep high nibble of address
   $0200.0000 or           ( bit adr' )             \ Offset to bitband alias space
   swap la+                ( aliasadr )             \ Merge bit number as 32-bit word offset
;
```

Reply:
what about the following usecase:

```\ set bit 1 of x
VARIABLE x
\ ...
1 x BITBAND
1 SWAP C!
```
In this case it is difficult to type the address of the bit that corresponds to the address of the variable.